src/mutex.rs

```diff
@@ -316,21 +318,77 @@ impl<T: 'static + ?Sized> Mutex<T> {
     /// # extern crate tokio;
     /// # use futures_locks::*;
     /// # use futures::{Future, IntoFuture, lazy};
-    /// # use tokio::executor::current_thread;
+    /// # use tokio::runtime::current_thread::Runtime;
     /// # fn main() {
     /// let mtx = Mutex::<u32>::new(0);
-    /// let r = current_thread::block_on_all(lazy(|| {
+    /// let mut rt = Runtime::new().unwrap();
+    /// let r = rt.block_on(lazy(|| {
     ///     mtx.with(|mut guard| {
     ///         *guard += 5;
     ///         Ok(()) as Result<(), ()>
-    ///     })
+    ///     }).unwrap()
     ///     .map(|_| assert_eq!(mtx.try_unwrap().unwrap(), 5))
     /// }));
     /// assert!(r.is_ok());
     /// # }
     /// ```
     #[cfg(feature = "tokio")]
-    pub fn with<F, B, R, E>(&self, f: F) -> oneshot::Receiver<Result<R, E>>
+    pub fn with<F, B, R, E>(&self, f: F)
+        -> Result<oneshot::Receiver<Result<R, E>>, SpawnError>
+        where F: FnOnce(MutexGuard<T>) -> B + Send + 'static,
+              B: IntoFuture<Item = R, Error = E> + 'static,
+              <B as IntoFuture>::Future: Send,
+              R: Send + 'static,
+              E: Send + 'static,
+              T: Send
+    {
+        let (tx, rx) = oneshot::channel::<Result<R, E>>();
+        tokio::executor::DefaultExecutor::current().spawn(Box::new(self.lock()
```

Comment:
Any reason why you're not just using `tokio_executor::spawn`? Iirc it uses DefaultExecutor under the hood

Reply:
just because `DefaultExecutor::spawn` is what Carl Lerche recommended in that issue that I linked.